config/webhook/kustomization.yaml

```diff
@@ -40,4 +35,8 @@ patchesJson6902:
         matchExpressions:
           - key: control-plane
             operator: NotIn
-            values: ["k8ssandra-operator"]
\ No newline at end of file
+            values: ["k8ssandra-operator"]
+
+# adding the objectSelector prevents the bootstrapping problem
+# where the mutation request for the operator pod would be 
+# sent before the operator pod is created
```

Comment:
OK, I've found a way to make it work, but there's definitely something weird here: `namePrefix` **already** replaces the service name in the webhook configs (as specified in the `nameReference` section of `kustomizeconfig.yaml`), so all the `replace` patches in this file are in fact **ignored**. You can observe that by setting any dummy content in `value:`, the resulting name will always be `k8ssandra-operator-webhook-service`.

Next, while `patchesJson6902` worked with `name: k8ssandra-operator-validating-webhook-configuration`, it looks like `patches` requires the pre-namePrefix name instead.

Putting it all together, I believe there should only be one patch in this file:
```yaml
patches:
  - target:
      group: admissionregistration.k8s.io
      version: v1
      name: mutating-webhook-configuration
      kind: MutatingWebhookConfiguration
    # adding the objectSelector prevents the bootstrapping problem
    # where the mutation request for the operator pod would be
    # sent before the operator pod is created
    patch: |-
      - op: add
        path: /webhooks/0/objectSelector
        value:
          matchExpressions:
            - key: control-plane
              operator: NotIn
              values: ["k8ssandra-operator"]
```
I've confirmed that this fixes the deployment issue.

Reply:
Yeah, I wonder if this worked for a while.. indeed that patch is enough to make the output the same default:

```console
➜  k8ssandra-operator git:(modern_tools) ✗ diff default-new.yaml default-old.yaml
12d11
<     cert-manager.io/inject-ca-from: k8ssandra-operator/k8ssandra-operator-serving-cert
30477d30475
<     cert-manager.io/inject-ca-from: k8ssandra-operator/k8ssandra-operator-serving-cert
30799d30796
<     cert-manager.io/inject-ca-from: k8ssandra-operator/k8ssandra-operator-serving-cert
30878d30874
<     cert-manager.io/inject-ca-from: k8ssandra-operator/k8ssandra-operator-serving-cert
30944d30939
<     cert-manager.io/inject-ca-from: k8ssandra-operator/k8ssandra-operator-serving-cert
31050d31044
<     cert-manager.io/inject-ca-from: k8ssandra-operator/k8ssandra-operator-serving-cert
31136d31129
<     cert-manager.io/inject-ca-from: k8ssandra-operator/k8ssandra-operator-serving-cert
31241d31233
<     cert-manager.io/inject-ca-from: k8ssandra-operator/k8ssandra-operator-serving-cert
33623d33614
<     cert-manager.io/inject-ca-from: k8ssandra-operator/k8ssandra-operator-serving-cert
33757d33747
<     cert-manager.io/inject-ca-from: k8ssandra-operator/k8ssandra-operator-serving-cert
➜  k8ssandra-operator git:(modern_tools) ✗
```

Reply:
> I've seen this happening after deleting a previous installation of the operator. We now have a mutating webhook that is registered in k8ssandra-operator to inject secrets, and if you delete the operator in an unclean way, it doesn't get unregistered but isn't accessible anymore. Was this a fresh cluster or did you previously perform an installation on it?

This is interesting. Maybe we should consider this an indication that we should split the webhook into a different binary and install in `system`... It seems quite dangerous that we can delete a deployment and this will prevent pod creation thereafter :/

Reply:
It does seem dangerous indeed. Isn't there a finalizer that we could add to do a proper cleanup before going into larger refactorings?

Reply:
I don't see how. A finalizer is used by our operator to ensure that child resources undergo custom cleanup before the parent resource is deleted.

In this case, you're deleting the operator which would do the custom cleanup, so what is going to remove the finalizer?

Reply:
There is probably some very strange world where the operator manages itself in some sense, watching pod resources and only allowing it's own deletion once it has deleted it's own webhook resource. It is a bit of a weird machine at that point, but there's probably no technical reason why it shouldn't work...